### PR TITLE
Even more defficiency prison changes

### PR DESCRIPTION
**Pull request by Konater**
It looked a little empty and it was one of the areas I wasn't happy with so I stole a bunch of more stuff to make it less empty
If you haven't guessed by now I have a dark floor tile obsession and I'm seeking help for it
perma3

### CODE REVIEW

File: code/game/machinery/doors/glasspoddoor.dm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/obj/machinery/door/poddoor/shutters/New()
+	..()
+	layer = 3.1
+
+/obj/machinery/door/poddoor/glass/preopen
+	icon_state = "pdoor_glass_opened"
+	density = 0
+
+/obj/machinery/door/poddoor/shutters/attackby(obj/item/weapon/C as obj, mob/user as mob)
```
Comment (ComicIronic) [inline code/game/machinery/doors/glasspoddoor.dm]: Bad copypaste, you should be changing the poddoor attackby logic instead.

File: code/game/machinery/doors/glasspoddoor.dm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	name = "Shutters"
+	icon = 'icons/obj/doors/rapid_pdoor.dmi'
+	icon_state = "pdoor_glass_closed"
+	var/sound_open = 'sound/machines/shutter_open.ogg'
+	var/sound_close = 'sound/machines/shutter_close.ogg'
+	opacity = 0
+
+/obj/machinery/door/poddoor/shutters/New()
+	..()
+	layer = 3.1
```
Comment (ComicIronic) [inline code/game/machinery/doors/glasspoddoor.dm]: Why

File: code/game/machinery/doors/glasspoddoor.dm [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		spawn(-1)
+			flick("pdoor_glass_open", src)
+			icon_state = "pdoor_glass_opened"
+			sleep(15)
+			density = 0
+			operating = 0
+			return
+	return
+
+/obj/machinery/door/poddoor/glass/open()
```
Comment (ComicIronic) [inline code/game/machinery/doors/glasspoddoor.dm]: REALLY BAD COPYPASTE

File: code/game/machinery/doors/glasspoddoor.dm [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	update_nearby_tiles()
+
+	if(operating == 1) //emag again
+		operating = 0
+	if(autoclose)
+		spawn(150)
+			autoclose()		//TODO: note to self: look into this ~Carn
+	return 1
+
+/obj/machinery/door/poddoor/glass/close()
```
Comment (ComicIronic) [inline code/game/machinery/doors/glasspoddoor.dm]: JESUS

File: code/game/objects/structures/fitness.dm [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	if(in_use)
+		user << "Its already in use - wait a bit."
+		return
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
```
Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm:53]: Make this a type list for easier editing.

File: code/game/objects/structures/fitness.dm [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+			animate(user, pixel_y = -2, time = 3)
+			sleep(3)
+			animate(user, pixel_y = -4, time = 3)
+			sleep(3)
+			playsound(user, 'sound/effects/spring.ogg', 60, 1)
+
+		playsound(user, 'sound/machines/click.ogg', 60, 1)
+		in_use = 0
+		user.pixel_y = 0
+		var/finishmessage = pick("You feel stronger!","You feel like you can take on the world!","You feel robust!","You feel indestructible!")
```
Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm:68]: Same idea.

**Comment by PJB3005** [comment]
Other than the code, :+1: looks really sick. The transition between light and dark feels a little too big though.

File: code/game/objects/structures/fitness.dm [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+			sleep(3)
+			animate(user, pixel_y = -2, time = 3)
+			sleep(3)
+			animate(user, pixel_y = -4, time = 3)
+			sleep(3)
+			playsound(user, 'sound/effects/spring.ogg', 60, 1)
+
+		playsound(user, 'sound/machines/click.ogg', 60, 1)
+		in_use = 0
+		user.pixel_y = 0
```
Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm]: It should not do this, direct pixel resetting is going to play hell with stuff like wheelchairs.

File: code/game/machinery/doors/glasspoddoor.dm [hunk 8]
```diff
@@ -0,0 +1,66 @@
+/obj/machinery/door/poddoor/glass
+	name = "Shutters"
```
Comment (PJB3005) [inline code/game/machinery/doors/glasspoddoor.dm]: Should make this not capitalized.

File: code/game/objects/structures/fitness.dm [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+/obj/structure/weightlifter
+	name = "Weight Machine"
+	desc = "Just looking at this thing makes you feel tired."
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "fitnessweight"
+	density = 1
+	anchored = 1
+
+/obj/structure/weightlifter/attack_hand(mob/user as mob)
```
Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm:80]: Holy fuck how much copypaste is IN this

File: code/game/objects/structures/fitness.dm [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+	desc = "A punching bag. Can you get to speed level 4???"
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "punchingbag"
+	anchored = 1
+	var/list/hit_sounds = list('sound/weapons/genhit1.ogg', 'sound/weapons/genhit2.ogg', 'sound/weapons/genhit3.ogg',\
+	'sound/weapons/punch1.ogg', 'sound/weapons/punch2.ogg', 'sound/weapons/punch3.ogg', 'sound/weapons/punch4.ogg')
+
+/obj/structure/punching_bag/attack_hand(mob/user as mob)
+	flick("[icon_state]2", src)
+	playsound(src.loc, pick(src.hit_sounds), 25, 1, -1)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: `src.`

Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm]: HE POSTED IT AGAIN

File: code/game/objects/structures/fitness.dm [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+/obj/structure/punching_bag/clown
+	name = "clown bop bag"
+	desc = "A bop bag in the shape of a goofy clown."
+	icon_state = "bopbag"
+
+/obj/structure/punching_bag/clown/attack_hand(mob/user as mob)
+	flick("[icon_state]2", src)
+	playsound(src.loc, pick(src.hit_sounds), 25, 1, -1)
+	playsound(src.loc, 'sound/items/bikehorn.ogg', 50, 1, -1)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: `src.`

File: code/game/objects/structures/fitness.dm [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+/obj/structure/stacklifter
+	name = "Weight Machine"
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Capitalization is off.

File: code/game/objects/structures/fitness.dm [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+	desc = "Just looking at this thing makes you feel tired."
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "fitnesslifter"
+	density = 1
+	anchored = 1
+
+/obj/structure/stacklifter/attack_hand(mob/user as mob)
+	if(in_use)
+		user << "Its already in use - wait a bit."
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: This should use `to_chat(user, message)`

File: code/game/objects/structures/fitness.dm [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+		return
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Use `forceMove

Remove `src.`

File: code/game/objects/structures/fitness.dm [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+	anchored = 1
+
+/obj/structure/stacklifter/attack_hand(mob/user as mob)
+	if(in_use)
+		user << "Its already in use - wait a bit."
+		return
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.dir = SOUTH
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Use `user.change_dir(SOUTH)`

File: code/game/objects/structures/fitness.dm [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+		user.Stun(4)
+		user.loc = src.loc
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/lifts = 0
+		while (lifts++ < 6)
+			if (user.loc != src.loc)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: `src.`

File: code/game/objects/structures/fitness.dm [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/lifts = 0
+		while (lifts++ < 6)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Sounds like a for loop would be better for this.

File: code/game/objects/structures/fitness.dm [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+			if (user.loc != src.loc)
+				break
+			sleep(3)
+			animate(user, pixel_y = -2, time = 3)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: No direct pixel offsets setting.

File: code/game/objects/structures/fitness.dm [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+			sleep(3)
+			animate(user, pixel_y = -4, time = 3)
```
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Again.

File: code/game/objects/structures/fitness.dm (hunk 10 above)
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm]: Hey look there's a second `src.` on this line.

**Comment by PJB3005** [comment]
The lighter area is obviously a basketball field so maybe you can make it use the darker floors too?

File: code/game/machinery/doors/glasspoddoor.dm [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+/obj/machinery/door/poddoor/glass/open()
+	if(operating == 1) //doors can still open when emag-disabled
+		return
+	if(!ticker)
+		return 0
+	if(!operating) //in case of emag
+		operating = 1
+	flick("pdoor_glass_open", src)
+	icon_state = "pdoor_glass_opened"
+	playsound(src.loc, sound_open, 100, 1)
```
Comment (ghost) [inline code/game/machinery/doors/glasspoddoor.dm]: Remove src

File: code/game/machinery/doors/glasspoddoor.dm [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+	return 1
+
+/obj/machinery/door/poddoor/glass/close()
+	if(operating)
+	//if(welded) //these are not airlocks.
+		return
+	operating = 1
+	flick("pdoor_glass_close", src)
+	icon_state = "pdoor_glass_closed"
+	playsound(src.loc, sound_close, 100, 1)
```
Comment (ghost) [inline code/game/machinery/doors/glasspoddoor.dm]: Remove src

File: code/game/machinery/doors/glasspoddoor.dm [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+	density = 1
+	update_nearby_tiles()
+
+	sleep(10)
+	operating = 0
+	return
```
Comment (ghost) [inline code/game/machinery/doors/glasspoddoor.dm]: Trailing return

File: code/game/machinery/doors/glasspoddoor.dm (hunk 20 above)
Comment (PJB3005) [inline code/game/machinery/doors/glasspoddoor.dm]: :+1:

File: code/game/machinery/doors/glasspoddoor.dm [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+	//if(welded) //these are not airlocks.
+		return
+	operating = 1
+	flick("pdoor_glass_close", src)
+	icon_state = "pdoor_glass_closed"
+	playsound(src.loc, sound_close, 100, 1)
+	density = 1
+	update_nearby_tiles()
+
+	sleep(10)
```
Comment (ghost) [inline code/game/machinery/doors/glasspoddoor.dm]: Not feeling the sleep() here, you sure this is the best way to do this?

File: code/game/objects/structures/fitness.dm [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+	desc = "It has a picture of a dumb looking station captain on it."
+
+/obj/structure/punching_bag/clown
+	name = "clown bop bag"
+	desc = "A bop bag in the shape of a goofy clown."
+	icon_state = "bopbag"
+
+/obj/structure/punching_bag/clown/attack_hand(mob/user as mob)
+	flick("[icon_state]2", src)
+	playsound(src.loc, pick(src.hit_sounds), 25, 1, -1)
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: Src

File: code/game/objects/structures/fitness.dm [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+/obj/structure/weightlifter/attack_hand(mob/user as mob)
+	if(in_use)
+		user << "Its already in use - wait a bit."
+		return
+	else
+		in_use = 1
+		icon_state = "fitnessweight-c"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: src and force move

File: code/game/objects/structures/fitness.dm [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+		var/image/W = image('icons/obj/fitness.dmi',"fitnessweight-w")
+		W.layer = MOB_LAYER + 0.1
+		overlays += W
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/reps = 0
+		user.pixel_y = 5
+		while (reps++ < 6)
+			if (user.loc != src.loc)
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: Src

File: code/game/objects/structures/fitness.dm [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+		user.Stun(4)
+		user.loc = src.loc
+		var/image/W = image('icons/obj/fitness.dmi',"fitnessweight-w")
+		W.layer = MOB_LAYER + 0.1
+		overlays += W
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/reps = 0
+		user.pixel_y = 5
+		while (reps++ < 6)
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: AAAAAAAAAAAAAA

**Comment by 9600bauds** [comment]
Is it possible to add a camera to make basketball matches televised like Dylan's courtroom? That would be sick

**Comment by PJB3005** [comment]
Add a central intercom too so the crew can be an audience for the inmates.

**Comment by Kurfursten** [comment]
This PR should replace the Vox seed vendor on the outpost

**Comment by ComicIronic** [comment]
It really shouldn't, atomic PRs.

**Comment by Kurfursten** [comment]
Map PRs have more leeway, map stuff is a pain due to merge priority, the map tools, etc.

Save everyone a headache

**Comment by PJB3005** [comment]
@Kurfurst I want to bet that it wouldn't conflict.

File: code/game/objects/structures/fitness.dm (hunk 27 above)
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: legit, this is
EXACTLY
what a for loop is for
a while loop is ONLY for when you don't know when it'll end

**Comment by Konater** [comment]
aw fug that's a lot of comments, 
also sorry kurf I started this before your thing got merged

File: code/game/objects/structures/fitness.dm (hunk 9 above)
Comment (Konater) [inline code/game/objects/structures/fitness.dm:80]: all of it

**Comment by Kurfursten** [comment]
@PJB3005 I hope for Konater you're right

@Konater That's what rebase is foooor

**Comment by Konater** [comment]
if you have a naming system for all of the sprites you could merge the pod door/shutters/glass pod door into one file right

File: code/game/machinery/doors/glasspoddoor.dm [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+	name = "shutters"
+	icon = 'icons/obj/doors/rapid_pdoor.dmi'
+	icon_state = "pdoor_glass_closed"
+	var/sound_open = 'sound/machines/shutter_open.ogg'
+	var/sound_close = 'sound/machines/shutter_close.ogg'
+	opacity = 0
+
+/obj/machinery/door/poddoor/glass/New()
+	..()
+	layer = 3.1
```
Comment (ghost) [inline code/game/machinery/doors/glasspoddoor.dm]: Magic numbers make me cry.

Comment (Konater) [inline code/game/machinery/doors/glasspoddoor.dm]: from poddoor.dm

> ```
>   layer = 3.3     //to override door.New() proc
> ```

it's in shutters.dm too

Comment (ComicIronic) [inline code/game/machinery/doors/glasspoddoor.dm]: But why are you setting it to 3.1?

File: code/game/objects/structures/fitness.dm [hunk 29]
```diff
@@ -55,18 +55,18 @@
 		user.visible_message("<B>[user] is [bragmessage]!</B>")
 		var/lifts = 0
 		while (lifts++ < 6)
-			if (user.loc != src.loc)
+			if (user.loc != loc)
 				break
 			sleep(3)
-			animate(user, pixel_y = -2, time = 3)
+			animate(user, pixel_y = pixel_y - 2, time = 3)
```
Comment (ComicIronic) [inline code/game/objects/structures/fitness.dm:59]: Should this not be user.pixel_y

Comment (Konater) [inline code/game/objects/structures/fitness.dm:59]: I dunno pjb gave m that line and my guy animated alright when I tested it

File: code/game/machinery/doors/glasspoddoor.dm (hunk 28 above)
Comment (Konater) [inline code/game/machinery/doors/glasspoddoor.dm]: In shutters it's 3.1 and that's what I copied and pasted

**Comment by Konater** [comment]
What's everyone thoughts on adding a preopen shutter to the cell entrances? This would mean someone won't just walk out of prison if an officer leaves the shutter open when he lets a prisoner through and the timer runs out. They would still be able to smash the windows to get into the other cells and get out that way but it would delay it a bit.

File: code/game/objects/structures/fitness.dm (hunk 29 above)
Comment (PJB3005) [inline code/game/objects/structures/fitness.dm:59]: Whoops it sure should be, my fault.

**Comment by ComicIronic** [comment]
Copypaste fixed when

**Comment by Konater** [comment]
If I pretend the problem doesn't exist will it go away?

**Comment by PJB3005** [comment]
Nope

**Comment by Konater** [comment]
well shit

**Comment by Konater** [comment]
Well I want to move the glass pod doors to poddoor.dm and it checks for initial opacity so I wouldn't have to do anything there (I know it works because I had to replace a few pre open poddoors that weren't being made opaque when closed) but I dunno how to go about getting all the different icons to show

**Comment by ComicIronic** [comment]
You should make the opening, open, closing, and closed iconstates into vars on the poddoor, and then change the opening and closing code to use them.

**Comment by ghost** [comment]
as it seems you don't know how to fix the while problem konater;

replace the var/reps and while(reps < 6) with
for(var/reps = 1; reps++; reps <= 6)

I probably cocked that up somehow as I wrote it from memory but whatever

File: code/game/machinery/doors/poddoor.dm [hunk 30]
```diff
@@ -16,11 +16,27 @@ var/list/poddoors = list()
 	animation_delay = 18
 	animation_delay_2 = 5
 
+	var/closedicon = "pdoor1"
+	var/openicon = "pdoor0"
+	var/closingicon = "pdoorc1"
+	var/openingicon = "pdoorc0"
+
 /obj/machinery/door/poddoor/preopen
 	icon_state = "pdoor0"
```
Comment (ComicIronic) [inline code/game/machinery/doors/poddoor.dm:25]: Replace this

**Comment by Konater** [comment]
I tried replacing both as you asked but got
<=: statement has no effect

**Comment by PJB3005** [comment]
Celt be should be using a `for (var/reps = 1 to 6)` instead.

**Comment by ghost** [comment]
That works?

**Comment by ghost** [comment]
Also I guess it's =<

File: code/game/objects/structures/fitness.dm [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.change_dir(SOUTH)
+		user.Stun(4)
+		user.forceMove(loc)
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/lifts = 0
+		while (lifts++ < 6)
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: Change this as well

Comment (Konater) [inline code/game/objects/structures/fitness.dm]: I'm so sorry but now I'm getting "missing left-hand argument to <." when I changed the two lines to
        for(var/lifts = 1; lifts++; lifts =< 6)

Comment (ghost) [inline code/game/objects/structures/fitness.dm]: Use what @PJB3005 said instead. Blame me for giving you bad code.

File: code/game/objects/structures/fitness.dm [hunk 32]
```diff
@@ -0,0 +1,116 @@
...
+	name = "Weight Machine"
+	desc = "Just looking at this thing makes you feel tired."
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "fitnessweight"
+	density = 1
+	anchored = 1
+
+/obj/structure/weightlifter/attack_hand(mob/user as mob)
+	if(in_use)
+		to_chat(user, "<span class='notice'>Its already in use - wait a bit.</span>")
```
Comment (ghost) [inline code/game/objects/structures/fitness.dm]: It's*

File: code/game/machinery/doors/poddoor.dm [hunk 33]
```diff
@@ -47,8 +63,8 @@ var/list/poddoors = list()
 	if ((src.density && (stat & NOPOWER) && !( src.operating )))
```
Comment (ghost) [inline code/game/machinery/doors/poddoor.dm]: I know this isn't technically your code but could you remove all the "src."s here, they're unnecessary.

File: code/game/objects/structures/fitness.dm [hunk 34]
```diff
@@ -0,0 +1,116 @@
...
+	desc = "It has a picture of a dumb looking station captain on it."
+
+/obj/structure/punching_bag/clown
+	name = "clown bop bag"
+	desc = "A bop bag in the shape of a goofy clown."
+	icon_state = "bopbag"
+
+/obj/structure/punching_bag/clown/attack_hand(mob/user as mob)
+	flick("[icon_state]2", src)
+	playsound(loc, pick(hit_sounds), 25, 1, -1)
```
Comment (Kurfursten) [inline code/game/objects/structures/fitness.dm]: instead of pasting this, you should ..()

File: code/game/machinery/doors/poddoor.dm (hunk 33 above)
Comment (Konater) [inline code/game/machinery/doors/poddoor.dm]: technically none of this is my code anyway

File: code/game/machinery/doors/poddoor.dm (hunk 30 above)
Comment (Konater) [inline code/game/machinery/doors/poddoor.dm:25]: With what though? I remember getting an error if I just put in openicon

Comment (ComicIronic) [inline code/game/machinery/doors/poddoor.dm:25]: Oh right, I thought this was a proc and not a type. You can leave this.

**Pull request merged**